### DISCUSSION TITLE
Clarification on order Parameter in /api/v2/setting/{id}/ and /api/v2/set/{id}/ APIs

### DISCUSSION THREAD

**Discussion by vanshaj-pahwa** [wger-project/wger#1858] [General]
Hi, could you please clarify what the `order` parameter represents and what its expected value should be in the request body for the `/api/v2/setting/{id}/` and `/api/v2/set/{id}/` APIs?

Additionally, how can we add a set for a particular exercise using API with weight, reps?

**Comment by rolandgeider** [comment]
hi! the order allows changing the order of sets (e.g. per drag and drop). To add an exercise over the api, you need to
* create a set
* create a setting and there set the `exercise_base` as well as the `weight` and `reps`.
* if you add a second setting to a set it will create a superset

Also note that the way the routines work is going to change massively once this pr is merged https://github.com/wger-project/wger/pull/1827  What are you trying to do?

**Comment by vanshaj-pahwa** [comment]
Hi! I am integrating the API with the UI where users can add weights and reps. From what I understand, we first need to create a set and then create a setting where we specify the `exercise_base`, `weight`, and `reps` and pass set as a object which we just created? Let me know if I got that right or if there's anything else I should consider.  Or could you please help how can we achieve this by passing `exercise_base`, `weight`, `reps` and `set_number`. Thanks!

**Reply by rolandgeider** [reply]
Sorry for the late reply! There is currently only one blocker stoping us from merging the new routines which changes all of this. The good thing is that there is now documentation explaining how to create routines over the API (which might be a bit more complex now):

https://fosstodon.org/@wger/113935014751279232
https://github.com/wger-project/docs/blob/feature/flexible-routines/docs/api/routines.rst
